src/hotspot/share/opto/superword.cpp

```diff
@@ -2237,396 +1992,65 @@ void SuperWord::apply_memops_reordering_with_schedule(Node_List& memops_schedule
-          in2 =  vector_opd(p, 2);
-        }
-        vn = VectorNode::make(opc, in1, in2, vlen, velt_basic_type(n));
-        vlen_in_bytes = vn->as_Vector()->length_in_bytes();
-      } else if (VectorNode::is_roundopD(n)) {
-        Node* in1 = vector_opd(p, 1);
-        Node* in2 = first->in(2);
-        assert(in2->is_Con(), "Constant rounding mode expected.");
-        vn = VectorNode::make(opc, in1, in2, vlen, velt_basic_type(n));
-        vlen_in_bytes = vn->as_Vector()->length_in_bytes();
-      } else if (VectorNode::is_muladds2i(n)) {
-        assert(n->req() == 5u, "MulAddS2I should have 4 operands.");
-        Node* in1 = vector_opd(p, 1);
-        Node* in2 = vector_opd(p, 2);
-        vn = VectorNode::make(opc, in1, in2, vlen, velt_basic_type(n));
-        vlen_in_bytes = vn->as_Vector()->length_in_bytes();
-      } else if (opc == Op_SignumF || opc == Op_SignumD) {
-        assert(n->req() == 4, "four inputs expected");
-        Node* in = vector_opd(p, 1);
-        Node* zero = vector_opd(p, 2);
-        Node* one = vector_opd(p, 3);
-        vn = VectorNode::make(opc, in, zero, one, vlen, velt_basic_type(n));
-        vlen_in_bytes = vn->as_Vector()->length_in_bytes();
-      } else if (n->is_Cmp()) {
-        // Bool + Cmp + CMove -> VectorMaskCmp + VectorBlend
-        continue;
-      } else if (n->is_Bool()) {
-        // Bool + Cmp + CMove -> VectorMaskCmp + VectorBlend
-        continue;
-      } else if (n->is_CMove()) {
-        // Bool + Cmp + CMove -> VectorMaskCmp + VectorBlend
-
-        BoolNode* bol = n->in(1)->as_Bool();
-        assert(bol != nullptr, "must have Bool above CMove");
-        Node_List* bool_pack = get_pack(bol);
-        assert(bool_pack != nullptr, "CMove must have matching Bool pack");
-
-        CmpNode* cmp = bol->in(1)->as_Cmp();
-        assert(cmp != nullptr, "must have cmp above CMove");
-        Node_List* cmp_pack = get_pack(cmp);
-        assert(cmp_pack != nullptr, "Bool must have matching Cmp pack");
-
-        Node* cmp_in1 = vector_opd(cmp_pack, 1);
-        Node* cmp_in2 = vector_opd(cmp_pack, 2);
-
-        Node* blend_in1 = vector_opd(p, 2);
-        Node* blend_in2 = vector_opd(p, 3);
-
-        VTransformBoolTest bool_test = _packset.get_bool_test(bool_pack);
-        BoolTest::mask test_mask = bool_test._mask;
-        if (bool_test._is_negated) {
-           // We can cancel out the negation by swapping the blend inputs.
-           swap(blend_in1, blend_in2);
-        }
-
-        // VectorMaskCmp
-        ConINode* test_mask_node  = igvn().intcon((int)test_mask);
-        BasicType bt = velt_basic_type(cmp);
-        const TypeVect* vt = TypeVect::make(bt, vlen);
-        VectorNode* mask = new VectorMaskCmpNode(test_mask, cmp_in1, cmp_in2, test_mask_node, vt);
-        phase()->register_new_node_with_ctrl_of(mask, p->at(0));
-        igvn()._worklist.push(mask);
-
-        // VectorBlend
-        vn = new VectorBlendNode(blend_in1, blend_in2, mask);
```

Comment:
Note: `Cmp -> Bool -> CMove`
`Cmp` -> `VTransformElementWiseVectorNode` (but empty, i.e. no C2 node generated)
`Bool` -> `VTransformBoolVectorNode` (creates `VectorMaskCmpNode`)
`CMove` -> `VTransformElementWiseVectorNode` (creates `VectorBlendNode`)
Note: the `swap(blend_in1, blend_in2);` happens in `SuperWordVTransformBuilder::build_edges_for_vector_vtnodes`, where we look for the input `VTransformBoolVectorNode` and check if its test is negated. I'm not perfectly happy with this, but maybe I can improve this in the future with some kind of "IGVN" optimization on the `VTransformNodes`.